src/journal/compress.c

```diff
@@ -112,7 +112,7 @@ int compress_blob_lz4(const void *src, uint64_t src_size,
         if (src_size < 9)
                 return -ENOBUFS;
 
-        r = LZ4_compress_limitedOutput(src, dst + 8, src_size, (int) dst_alloc_size - 8);
+        r = LZ4_compress_limitedOutput(src, (char*)dst + 8, src_size, (int) dst_alloc_size - 8);
```

Comment:
Actually I tried `uint8_t` ;) It gives a warning about mismatched signedness. LZ4 uses `char*`, and `char` is signed on amd64, hence I got the warning. But using `int8_t` wouldn't work either, because on other arches `char` is unsigned. So `char*` is the only thing that works cleanly without too much trouble.